### DISCUSSION TITLE
Potential Batch Biases in FindMultiModalNeighbors() with Corrected Embeddings

### DISCUSSION THREAD

**Discussion by GhamsariReza** [satijalab/seurat#8777] [Q&A]
Hello everyone,

I've been using FindMultiModalNeighbors() for integrating ATAC and RNA modalities and it works great when I am dealing with one sample. I have some concerns regarding its behaviour with batch effects, when using harmony-corrected embeddings from RNA and ATAC data.

 In scenarios with multiple samples where batch effects are prevalent, many of us, as noted in comment #6089, provide harmony or other methods corrected embeddings as input to build the WSNN graph. From my understanding, FindMultiModalNeighbors()  uses raw counts (uncorrected data) for weighting each modality based on neighbourhood similarity. Here’s where my concern lies: could FindMultiModalNeighbors() inadvertently reintroduce batch biases when calculating weights for the modalities? This would imply a potential step backwards in effectively correcting batch effects.

Could anyone clarify if my understanding is correct or if I’m missing something? Any insights or suggestions on how to better handle this situation would be greatly appreciated!

Thank you!

**Comment by cynthier** [comment]
According to the results, I think it would. After Harmony/Integration, the RNA data of multiple samples aligned well, so does ATAC data. However, when using WNN to integrate RNA and ATAC data, the batch effects among samples came back again, where cells clustered based on samples.
